### ISSUE TITLE
Rule: Fields must be qualified with table name - does not work

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#874]
> All buffer references (including database tables, temp-tables, and buffers) must be fully qualified.

Just spent quite some tracking down a relative of this piece of code which I thought was covered by this rule

````
DEFINE TEMP-TABLE tt NO-UNDO
   FIELD ifoo AS INTEGER
   .

FUNCTION bar RETURNS LOGICAL (
   i_ifoo AS INTEGER
):

   DEFINE BUFFER bu FOR tt.

   FIND bu NO-ERROR.
   IF i_ifoo > 1 THEN
      MESSAGE ifoo. // this is a mistake

   RETURN TRUE.

END FUNCTION.

bar(2).
````
The ifoo mistake is not flagged.

**Comment by gquerret** [comment]
This rule is flagged as `Compiler warning`, so it just reports all those warnings in the SonarQube database. I've used your example in a test project, and I can see the issue:
![image](https://user-images.githubusercontent.com/6498723/113099586-77a3b100-91fa-11eb-9e98-99ac05e45142.png)

However, this is not reported in SonarLint.

**Comment by stefandrissen** [comment]
Aha - so I need to add requireFullNames when using PCTcompile?

**Comment by gquerret** [comment]
Yes, the `require*` attributes in PCTCompile enable those warnings, and generate `.warnings` files in the `.pct` directory. Those files are then read by SonarQube.

**Comment by gquerret** [comment]
I assume that it worked in your environment. Please re-open in case of problem.
